docs/building-rpms-from-source.md

```diff
@@ -0,0 +1,166 @@
+# Building RPMs from source with mkosi
+
+If you want to build and RPM from source and install it within a mkosi
+image, you can do that with mkosi itself without using `mock`. The steps
+required are as follows:
+
+1. Install `Requires` dependencies in the image
+2. Install `BuildRequires` dependencies in the build overlay
+3. Install dynamic `BuildRequires` dependencies in the build overlay
+4. Build the RPM with `rpmbuild`
+5. Install the built rpms in the image
+
+In the following examples, we'll use mkosi itself and it's Fedora RPM
+spec as an example.
+
+To keep things snappy, we execute the first 3 steps in a prepare
+script so that they're cached on subsequent runs of mkosi if the
+`Incremental=` setting is enabled.
+
+First, we need access to the upstream sources and the RPM spec and
+related files. These can be mounted into the current working directory
+when running mkosi scripts by using the `BuildSources=` setting. For
+example, in `mkosi.local.conf`, we could have the following settings:
+
+`mkosi.local.conf`:
+
+```conf
+[Content]
```

Comment:
This might look weird as separate paragraphs, how about
```suggestion
```conf
# mkosi.local.conf
[Content]
```